mantidproject/mantid
# DirectILLDiagnostics: enable instrument specific diagnostics thresholds

**Description of work.**

This PR enables the setting of threshold parameters for noisy background and elastic peak diagnostics via the IPF. The IPF of IN6 was modified accordingly as per request from the instrument scientist.

Unrelated: While touching the IPFs, a few additional checks for [`MergeRuns`](http://docs.mantidproject.org/nightly/algorithms/MergeRuns-v1.html) were implemented for IN4 and IN6.

**Report to:** Björn Fåk.

**To test:**

Make sure you are using the correct IPF, not a random one downloaded during the IDF update at MantidPlot startup. Set the logging level to information, and check that the unit test data directory is in MantidPlot's data search directories. Run the following snippet: 

```python
DirectILLCollectData(Run='ILL/IN4/084446.nxs', OutputWorkspace='ws', OutputEPPWorkspace='epps')
DirectILLDiagnostics(InputWorkspace='ws', OutputWorkspace='diagnostics', EPPWorkspace='epps', SubalgorithmLogging='Logging ON')
```

From the logs, observe that the last call to `MedianDetectorTest` (the input workspace name has references to 'elastic_peak') was called with LowThreshold = 0.1. This is the default value since the instrument above is IN4. Repeat with IN6 data:
```python
DirectILLCollectData(Run='ILL/IN6/164192.nxs', OutputWorkspace='ws', OutputEPPWorkspace='epps')
DirectILLDiagnostics(InputWorkspace='ws', OutputWorkspace='diagnostics', EPPWorkspace='epps', SubalgorithmLogging='Logging ON')
```

Check that `MedianDetectorTest` was called with LowThreshold=0.45 as it states in the IPF.

Fixes #22404.

Does this update require release notes?

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
